### PR TITLE
[#286] prompt to move the app to `Applications` folder in mac if launched from somewhere else

### PR DESCRIPTION
**Pull request by prakashchokalingam**
https://github.com/manojVivek/responsively-app/issues/286

Screenshot 2020-07-10 at 12 17 42 AM

### CODE REVIEW

File: desktop-app/app/app-move-to-applications.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const response: number = dialog.showMessageBoxSync({
+    type: 'error',
+    message: 'Move to Applications folder?',
+    buttons: ['Move', 'Cancel'],
+  });
+  return response === 0;
+};
+
+const conflictHandler = (conflictType, dialog) => {
+  if (conflictType === 'exists' || conflictType === 'existsAndRunning') {
```
Comment (manojVivek) [inline desktop-app/app/app-move-to-applications.js:21]: @prakashchokalingam What will happen if the `conflictType` is something else? 
Would it fail the application from starting?

File: desktop-app/app/app-move-to-applications.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  https://www.electronjs.org/docs/api/app#appmovetoapplicationsfolderoptions-macos
+*/
+
+// @flow
+
+let conflictMessage = 'An app of this name already exists';
+const moveFailedMessage = 'Something went wrong while moving application';
+
+const confirmMove: boolean = dialog => {
+  const response: number = dialog.showMessageBoxSync({
```
Comment (manojVivek) [inline desktop-app/app/app-move-to-applications.js:12]: Can you please add a `detail` field here with the string `Responsively App should be in the Applications folder to be able to run correctly.`

**Comment by manojVivek** [comment]
@prakashchokalingam Awesome work on this one. 🙌  
Can you please add some screenshots/gif of how the flow works?

**Comment by manojVivek** [comment]
@prakashchokalingam Can you please add a comment on the #286. So that GitHub will allow me to assign that issue to you?  

Also, join us on Gitter, if you have any questions to discuss.  - https://gitter.im/responsively-app/

File: desktop-app/app/app-move-to-applications.js (hunk 1 above)
Comment (prakashchokalingam) [reply desktop-app/app/app-move-to-applications.js:21]: As per the documentation the conflict will occur only if the app of same name already exists in applications.The above are the only two conflict type electron will throw.

https://www.electronjs.org/docs/api/app#appmovetoapplicationsfolderoptions-macos

Any other issue will be thrown as error, that we are catching in the try catch block. 

For safe: we can also throw a custom error if conflictType condition is not satisfied.

File: desktop-app/app/app-move-to-applications.js (hunk 2 above)
Comment (prakashchokalingam) [reply desktop-app/app/app-move-to-applications.js:12]: Sure will do that.

**Comment by prakashchokalingam** [comment]
@manojVivek I have one doubt, this change will prompt the end user always, if he tries to run this app from somewhere else.

What if he doesn't want to move to applications?

Can we add a new button '**Don't ask me again**' and store the preference? Which will save the end user from this nudge.

**Comment by manojVivek** [comment]
> Can we add a new button '**Don't ask me again**' and store the preference? Which will save the end user from this nudge.

@prakashchokalingam This is a good one, let's add it.

**Comment by prakashchokalingam** [comment]
> @prakashchokalingam Awesome work on this one. 🙌
> Can you please add some screenshots/gif of how the flow works?

Have added the screenshots to the description.

**Comment by manojVivek** [comment]
> > Can we add a new button '**Don't ask me again**' and store the preference? Which will save the end user from this nudge.
> 
> @prakashchokalingam This is a good one, let's add it.

Hey @prakashchokalingam, Do you want to add this in so that this PR can be merged? :)

**Comment by prakashchokalingam** [comment]
@manojVivek give me few days will complete this PR along with 'Don't ask me again'.

**Comment by manojVivek** [comment]
@prakashchokalingam No worries, was just checking. 👍🏼

**Comment by jjavierdguezas** [comment]
@prakashchokalingam if I understand it well, this PR was OK and we were just waiting for an improvement... Why did you close it? It has an issue? If this is OK, we can merge it and move the improvement to another issue
cc: @manojVivek

**Comment by manojVivek** [comment]
@jjavierdguezas I'll raise a new PR with the improvement made along with this. 👍🏼

**Pull request closed without merging**